Signals/Domain/MissingValuePolicy/FirstOrderMissingValuePolicy.cs

```diff
@@ -40,24 +49,30 @@ public class FirstOrderMissingValuePolicy<T> : MissingValuePolicy<T>
 
         private Quality GetMinQuality(Quality a, Quality b)
         {
-            // TODO should we add numbers to Quality? Those would probably go into Dto and suggest some impl...
             if (a == Quality.None || b == Quality.None)
                 return Quality.None;
-            return (Quality)Math.Max((int)a, (int)b);
+
+            return Enum
+                .GetValues(typeof(Quality))
+                .Cast<Quality>()
+                .Last(q => q == a || q == b);
```

Comment:
i.e. - it is enforced/tested :) Note that our implementation doesn't really matter. If students will manage to implement this without enforcing the order - no problem :)